### PR TITLE
Correcting and improving documentation of TailChoppingRoutingLogic, T…

### PR DESCRIPTION
**Pull request by dodnert**
Correcting and improving documentation of TailChoppingRoutingLogic, TailChoppingPool, and TailChoppingGroup (issue #17640). Documentation changes only; no code changes.

### CODE REVIEW

File: akka-actor/src/main/scala/akka/routing/TailChopping.scala [hunk 1]
```diff
@@ -19,15 +19,20 @@ import akka.util.Helpers.ConfigOps
 import scala.util.Random
 
 /**
- * Sends the message to a first, random picked, routee,
- * then wait a specified `interval` and then send to a second, random picked, and so on till one full cycle.
+ * The routees are randomly ordered. The message is sent to the first routee. If no response is received
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/routing/TailChopping.scala]: perhaps even: "The routees are randomly shuffled for each routed message."

**Comment by patriknw** [comment]
OK TO TEST

**Comment by patriknw** [comment]
LGTM

File: akka-actor/src/main/scala/akka/routing/TailChopping.scala (hunk 1 above)
Comment (ktoso) [inline akka-actor/src/main/scala/akka/routing/TailChopping.scala]: I like that rewording :+1:

File: akka-actor/src/main/scala/akka/routing/TailChopping.scala [hunk 2]
```diff
@@ -19,15 +19,20 @@ import akka.util.Helpers.ConfigOps
...
+ * is sent to the sender.
  *
  * @param scheduler schedules sending messages to routees
  *
  * @param within expecting at least one reply within this duration, otherwise
  *   it will reply with [[akka.pattern.AskTimeoutException]] in a [[akka.actor.Status.Failure]]
  *
- * @param interval duration after which next routee will be picked
+ * @param interval duration after which the next routee will be sent the same message
+ *   sent to the previous routee
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/routing/TailChopping.scala]: `duration after which the message will be sent to the next routee`

File: akka-actor/src/main/scala/akka/routing/TailChopping.scala [hunk 3]
```diff
@@ -83,8 +88,14 @@ private[akka] final case class TailChoppingRoutees(
...
- * A router poll that sends the message to a first, random picked, routee,
- * then wait a specified `interval` and then send to a second, random picked, and so on till one full cycle..
+ * A router pool with retry logic, intended for cases where a return message is expected in
+ * response to a message sent to the routee. As each message is sent to the routing pool, the
+ * routees are randomly ordered. The message is sent to the first routee. If no response is received
+ * before the `interval` has passed, the same message is sent to the next routee. This process repeats
+ * until either a response is received from some routee, the routees in the pool are exhausted, or
+ * the `within` duration has passed since the first send. If no routee sends
+ * a response in time, a [[akka.actor.Status.Failure]] wrapping a [[akka.pattern.AskTimeoutException]]
+ * is sent to the sender.
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/routing/TailChopping.scala]: Good improvements, however it does not explain the *why* we're doing this.

Let's also add something like this while we're fixing docs here:

```
The goal of this router is to decrease tail latencies ("chop off the tail", hence the name)
in cases where multiple routees can perform the same piece of work, and may sometimes
act much slower than usually. In this case, sending the additional work requests (also 
known as "backup requests") allows to decrease total response time considerably 
as it's less probably that multiple workers are expiriencing slowness at the same time.

This technique is explained in depth in Jeff Dean's presentation on 
<a href="http://static.googleusercontent.com/media/research.google.com/en//people/jeff/Berkeley-Latency-Mar2012.pdf">Achieving Rapid Response Times in Large Online Services</a>.
```

Comment (patriknw) [inline akka-actor/src/main/scala/akka/routing/TailChopping.scala]: :+1:

**Comment by ktoso** [comment]
Thanks, LGTM after the additional section is added :)

**Comment by dodnert** [comment]
Comments from Patrik and Konrad received so far have been incorporated.

**Comment by dodnert** [comment]
I'm new to this process. How do I know when this branch has "passed" the review process? Once it has passed, I am supposed to commit the changes to the master branch and push them to my forked repository on GitHub. Is there anything else I need to do?

**Comment by patriknw** [comment]
@dodnert this looks good now, so please squash the commits into one commit and force push to same docfix-17640-dodnert branch to update this PR. Then we will merge it. Thanks

**Comment by ktoso** [comment]
Yup, looks good! Please squash into one commit (using `git rebase --interactive`) and we'll merge :smile:

**Comment by patriknw** [comment]
Thanks for contributing!

**Pull request merged**